### DISCUSSION TITLE
Timezone setting on a postgres container level (not db level)

### DISCUSSION THREAD

**Discussion by MichaluxPL** [cloudnative-pg/cloudnative-pg#492] [Ideas]
Is there a way to set timezone on a container level ? For instance inject env variable TZ="XXXX/XXXX" ?
I know I can set timezone for PostgreSQL db itself, but that doesn't influence for instance backup Creation Timestamp. As a result creation timestamp does not correspond to Started At/Stopped At times. That also influences PITR times.
If it isn't than it would be good to enable podtemplate customization, jest like in Pooler CRD.

**Comment by sxd** [comment]
hi @MichaluxPL 

This sounds like a good idea, I'll turn this into a discussion so we can create issues that will be turn into development.
Do you have a proposal of how you think it should work?

Regards!

**Reply by MichaluxPL** [reply]
I think there should be a dedicated parameter in manifest (maybe in bootstrap section, or dedicated container section ?) the should accept standard timezone format (just like for postgres). Setting it should update accordingly /etc/localtime link (pointing to /usr/share/zoneinfo/XXX). That should do the trick.

**Reply by sxd** [reply]
@MichaluxPL 

The containers are immutable, meaning that nothing should change their configuration after it was created, that's the main idea there, that's why you can change the TZ in PostgreSQL using `spec.postgresql.parameters.timezone`.

Now related to the timezone of the backup, probably we can look into pass that argument to the Barman Cloud tools, since we know the timezone that it was configured, probably that will make sense and will help to your issue, what do you think?
Probably @leonardoce or @gbartolini and give us a hint on this?

**Reply by MichaluxPL** [reply]
Passing TZ to Barman Cloud would be a good start, I guess. On the other hand, that still doesn't solve the problem of all other logs generated by the container (CNPG manager included ?) - it will still have the wrong timestamp. From the diagnostic point of view this is crucial to have one TZ setting for the whole system (K8s, Operator, PODs, containers etc.). Otherwise debugging can give You a strong headache.
Concerning immutability, IMHO timezone setting shouldn't impact that (yes, my proposed idea pobably would :) ). Creating the container, for instance, You inject manager to the container (so You kind of change it). Maybe we could change timezone in a similar way ? Maybe PGSQL image itself should be prepared in a way, that allows timezone setting, not impacting container immutability ? So the idea would be to set a correct timezone during POD/Container creation and of course not change it during its lifetime. It could be changed only by recreating the POD (just like for the CPU/Memeory parameters ?).

**Reply by MichaluxPL** [reply]
Hi,
Any news regarding this one ?
Can we move it forward ?
